File: Android project/RoomDataBase/app/src/main/java/com/mirkamol/roomdatabase/database/UserRepository.java
package com.mirkamol.roomdatabase.database;

import android.app.Application;

import com.mirkamol.roomdatabase.manager.RoomManager;
import com.mirkamol.roomdatabase.model.User;

import java.util.List;

public class UserRepository {
    private UserDao userDao;
    public UserRepository(Application application){
        RoomManager db = RoomManager.getDatabase(application);
        userDao = db.usersDao();
    }

    public List<User> getUsers(){
        return userDao.getUsers();
    }
    public void saveUser(User user){
        userDao.saveUser(user);
    }
}

File: Android project/OgabekDevBestPractise/app/src/main/java/com/mirkamol/ogabekdevbestpractise/database/UserRepository.kt
package com.mirkamol.ogabekdevbestpractise.database

import android.app.Application
import com.mirkamol.ogabekdevbestpractise.helper.Logger

import com.mirkamol.ogabekdevbestpractise.manager.RoomManager


class UserRepository(application: Application) {
    val TAG:String = UserRepository::class.java.simpleName

    private val db = RoomManager.getDatabase(application)
    private val userDao:UserDao = db!!.userDao()

    fun getUsers():List<UserDB>{
        Logger.d(TAG, "Done")

        return userDao.getUsers()
    }

    fun saveUser(userDB: UserDB){
        Logger.d(TAG, "Saved")
        userDao.saveUser(userDB)
    }

    fun deleteUsers(){
        Logger.d(TAG, "Database cleared")

        userDao.deleteAllUsers()
    }
}